op-node/p2p/sync.go

```diff
@@ -95,17 +103,49 @@ type syncResult struct {
 }
 
 type peerRequest struct {
-	num uint64
-
-	complete *atomic.Bool
+	num        uint64
+	rangeReqId uint64
 }
 
 type inFlightCheck struct {
-	num uint64
-
+	num    uint64
 	result chan bool
 }
 
+type inFlight struct {
+	requests map[uint64]bool
+	mu       sync.Mutex
+}
+
+func newInFlight() *inFlight {
+	return &inFlight{
+		requests: make(map[uint64]bool),
+	}
+}
+
+func (s *inFlight) set(key uint64, value bool) {
+	s.mu.Lock()
+	s.requests[key] = value
+	s.mu.Unlock()
+}
+
+func (s *inFlight) get(key uint64) bool {
+	s.mu.Lock()
+	value, exists := s.requests[key]
+	s.mu.Unlock()
+	if !exists {
+		return false
+	} else {
+		return value
+	}
```

Comment:
nit: isn't this just:
```suggestion
    return exists && value
```

For that matter the default value of a bool in go is `false` so we could just return `s.requests[key]` directly and always get the right answer.

Reply:
Great point. Updated in https://github.com/ethereum-optimism/optimism/pull/10063/commits/32f9860c18cbd262d9d3d224b2ecbe7f925149b7